### ISSUE TITLE
Trailing slash op het endpoint van de FhirClient geeft problemen voor de medicatie use case

### ISSUE THREAD

**Issue by xmicix** [FirelyTeam/fhir-net-api#1094]
Hallo,

In de constructor van de FhirClient class wordt een trailing slash toegevoegd aan het endpoint als deze er niet staat. 
`    if (!endpoint.OriginalString.EndsWith("/"))
                endpoint = new Uri(endpoint.OriginalString + "/");
`

en Endpoint is een Private Set Property

        // The default endpoint for use with operations that use discrete id/version parameters
        // instead of explicit uri endpoints. This will always have a trailing "/"
`             public Uri Endpoint
        {
            get;
            private set;
        }
`

Maar voor de MedicationTreatment use case van MedMij horen wij de volgende calls te kunnen maken

http://touchstone.aegis.net:55953/fhir3-i2?medicationtreatment=urn:oid:2.16.840.1.113883.2.4.3.11.999.77.1.1|MBH_113_1&_include=MedicationRequest:medication&_include=MedicationDispense:medication&_include=MedicationStatement:medication

Echter door de werking van de API kunnen wij onmogelijk aan deze test voldoen.

Dit was tot voor kort geen probleem maar de testen op touchstone zijn recent aangepast.

https://bits.nictiz.nl/browse/MM-439?jql=text%20~%20%22medicationtreatment%22

Met of zonder trailing / krijgen we wel dezelfde data terug overigens. 

Hoort/kan dit aangepast te worden?

**Comment by mbaltus** [comment]
According to several HTTP definitions I've reviewed, it should not make a difference whether you use <endpoint>?<query> or <endpoint>/?<query>. So even though the FHIR specification shows the query on the base without the trailing slash, is should not be a problem that the FhirClient adds it. There is probably a good reason why the API adds the slash, but I leave it up to the developers to answer that.

**Comment by MichaelTimmermans** [comment]
I agree, it should not matter. And it doesn't matter response wise. We get the responses we need either way. Still wont harm anyone to not have a mandatory trailing slash.

**Comment by mbaltus** [comment]
Well... that last remark is tricky. It could introduce a lot of extra checks in other places in the code, both in the FhirClient and on the server side, which could potentially lead to bugs. That's why I said I'd let the developers answer the why. :)

**Comment by cknaap** [comment]
It would add more logic to everywhere we build the full path (``if (no '/' and no '?') add '/'``), so it is easier to do it in only one place.
